examples/budget_project/main.tf

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+provider "google" {
+  version = "~> 3.3.0"
+}
+
+provider "google-beta" {
+  version = "~> 3.3.0"
+}
+
+provider "null" {
+  version = "~> 2.1"
+}
+
+provider "random" {
+  version = "~> 2.2"
+}
+
+resource "random_string" "suffix" {
+  length  = 4
+  special = false
+  upper   = false
+}
+
+module "budget_project" {
+  source            = "../../"
+  name              = "budget-project-${random_string.suffix.result}"
+  random_project_id = true
+  org_id            = var.org_id
+  folder_id         = var.folder_id
+  billing_account   = var.billing_account
+  budget_amount     = var.budget_amount
+}
+
+
+# An additional budget with more options
+resource "google_pubsub_topic" "budget" {
+  name    = "budget-topic-${random_string.suffix.result}"
+  project = var.pubsub_project_id
```

Comment:
Just go ahead and put this in the project itself.

Reply:
Done